File: SourceCode/Debug Helpers/open.fda.gov/ApiDalc/DataObjects/DeviceEventPatientData.vb
#Region " Imports "

Imports ApiDalc.DataObjects
Imports Newtonsoft.Json.Linq ' OpenSource

#End Region

Namespace DataObjects

    Public Class DeviceEventPatientData

#Region " Properties "

        Public Property PatientSequenceNumber As String
        Public Property DateReceived As String
        Public Property SequenceNumberTreatment As String
        Public Property SequenceNumberOutcome As New List(Of String)
        ' Public Property MdrText As New List(Of DeviceEventMdrTextData)

#End Region

#Region " Public Methods "

        ''' <summary>
        ''' Convert JSON Data
        ''' </summary>
        ''' <param name="jsondata">JSON Object</param>
        ''' <returns>List(Of DeviceEventPatientData)</returns>
        ''' <remarks></remarks>
        Public Shared Function CnvJsonDataToList(jsondata As JArray) As List(Of DeviceEventPatientData)

            Dim result As New List(Of DeviceEventPatientData)

            For Each obj In jsondata

                Dim tmp As New DeviceEventPatientData

                With tmp

                    .PatientSequenceNumber = obj("patient_sequence_number")
                    .DateReceived = obj("date_received")
                    .SequenceNumberTreatment = obj("sequence_number_treatment")
                    For Each itm In obj("sequence_number_outcome")
                        .SequenceNumberOutcome.Add(itm)
                    Next

                    '.MdrText = DeviceEventMdrTextData.CnvJsonDataToList(obj("mdr_text"))

                End With

                result.Add(tmp)

            Next

            Return result

        End Function

#End Region

    End Class

End Namespace

File: SourceCode/AspCoreVersion/src/ShopAware.Core/DataObjects/DeviceEventPatientData.cs
using System.Collections.Generic;
using Newtonsoft.Json.Linq;

namespace ShopAware.Core
{
    namespace DataObjects
    {
        public class DeviceEventPatientData
        {
            #region Properties

            public string PatientSequenceNumber { get; set; }

            public string DateReceived { get; set; }

            public string SequenceNumberTreatment { get; set; }

            public List<string> SequenceNumberOutcome { get; set; }

            #endregion

            #region Constructors

            public DeviceEventPatientData()
            {
                // VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
                SequenceNumberOutcome = new List<string>();
            }

            #endregion

            #region Public Methods

            /// <summary>
            ///     Convert JSON Data
            /// </summary>
            /// <param name="jsondata">JSON Object</param>
            /// <returns>List(Of DeviceEventPatientData)</returns>
            /// <remarks></remarks>
            public static List<DeviceEventPatientData> CnvJsonDataToList(JArray jsondata)
            {
                var result = new List<DeviceEventPatientData>();

                foreach (var obj in jsondata)
                {
                    var tmp = new DeviceEventPatientData();

                    tmp.PatientSequenceNumber = (obj["patient_sequence_number"]).ToString();
                    tmp.DateReceived = (obj["date_received"]).ToString();
                    tmp.SequenceNumberTreatment = (obj["sequence_number_treatment"]).ToString();
                    foreach (var itm in obj["sequence_number_outcome"])
                    {
                        tmp.SequenceNumberOutcome.Add((itm).ToString());
                    }

                    //.MdrText = DeviceEventMdrTextData.CnvJsonDataToList(obj("mdr_text"))

                    result.Add(tmp);
                }

                return result;
            }

            #endregion
        }
    }
}